python/fledge/plugins/south/mqtt_sparkplug/mqtt_sparkplug.py

```diff
@@ -215,41 +220,61 @@ def plugin_register_ingest(handle, callback, ingest_ref):
 
 # MQTT Server Connection Callback
 # The callback for when the client receives a CONNACK response from the server.
-def on_connect(client, userdata, flags, rc):        
+def on_connect(client, userdata, flags, rc):
     # Subscribing in on_connect() means that if we lose the connection and
     # reconnect then subscriptions will be renewed.
     global _topic
     global _serverURL
     _LOGGER.info('{} plugin has connected to {}.'.format(_PLUGIN_NAME, str(_serverURL)))
-    
+
     client.subscribe(_topic)
     _LOGGER.info('{} plugin has subscribed to {}.'.format(_PLUGIN_NAME, str(_topic)))
 
-    
-async def save_data(data):    
+
+async def save_data(data):
     async_ingest.ingest_callback(c_callback, c_ingest_ref, data)
 
 
 # MQTT Message Received Callback
 # The callback for when a PUBLISH message is received from the server.
 def on_message(client, userdata, msg):
     global _callback_event_loop
+    global _assetName
     try:
-        inbound_payload = sparkplug_b_pb2.Payload()
-        inbound_payload.ParseFromString(msg.payload)
-        time_stamp = utils.local_timestamp()
-
         if _callback_event_loop is None:
-                _LOGGER.debug("Message processing event doesn't yet exist - creating new event loop.")
-                asyncio.set_event_loop(asyncio.new_event_loop())
-                _callback_event_loop = asyncio.get_event_loop()
-
-        for metric in inbound_payload.metrics:
+            _LOGGER.debug("Message processing event doesn't yet exist - creating new event loop.")
+            asyncio.set_event_loop(asyncio.new_event_loop())
+            _callback_event_loop = asyncio.get_event_loop()
+        time_stamp = utils.local_timestamp()
+        try:
+            string_data = msg.payload.decode('utf-8')
+            dict_data = json.loads(string_data)
+            data = {
+                'asset': _assetName,
+                'timestamp': time_stamp,
+                'readings': dict_data if isinstance(dict_data, dict) else {"value": string_data}
+            }
+            _callback_event_loop.run_until_complete(save_data(data))
+        except UnicodeDecodeError:
+            # Serialized data case with protobuf
+            inbound_payload = sparkplug_b_pb2.Payload()
+            inbound_payload.ParseFromString(msg.payload)
+            for metric in inbound_payload.metrics:
+                data = {
+                    'asset': metric.name,
+                    'timestamp': time_stamp,  # metric.timestamp
+                    'readings': {
+                        "value": metric.float_value,
```

Comment:
For sinusoid via MQTT north sparkplug B this should be enough to bring it back to life.

However, We should target a minimum of in next branch

```
# Define the message payload with different metric types
message = {
    "metrics": [
        {
            "name": "temperature",
            "value": 22.5,  # Float
            "timestamp": int(time.time()),
            "type": "float"
        },
        {
            "name": "humidity",
            "value": 55,  # Integer
            "timestamp": int(time.time()),
            "type": "integer"
        },
        {
            "name": "status",
            "value": "active",  # String
            "timestamp": int(time.time()),
            "type": "string"
        },
        {
            "name": "powerOn",
            "value": True,  # Boolean
            "timestamp": int(time.time()),
            "type": "boolean"
        }
    ]
}
```

May be we can also support float array, dict which Fledge supports.

Reply:
We have another JIRA FOGL-9198 to map Fledge Support types